libraries/lib-string-utils/FromChars.cpp

```diff
@@ -0,0 +1,190 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file FromChars.cpp
+ @brief Define functions to convert numeric types to string representation.
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#include "FromChars.h"
+
+#include "3party/fast_float.h"
+
+#include <limits>
+#include <algorithm>
+
+namespace
+{
+// Converting c to unsigned here helps to eliminate one check 
+unsigned digitToInt(char c) noexcept
+{
+   return unsigned(c - '0');
```

Comment:
`unsigned(c) - '0'` ?

I'm paranoid of signed char now.

Reply:
I'm pretty sure that this will generate the same code because of integer promotion rules. https://godbolt.org/z/4T8cfsMej

Can do it you way, no probs :-)